jaxrs/src/main/java/org/killbill/billing/jaxrs/util/Context.java

```diff
@@ -59,19 +59,15 @@ public CallContext createCallContextNoAccountId(final String createdBy, final St
 
     public CallContext createCallContextWithAccountId(final UUID accountId, final String createdBy, final String reason, final String comment, final ServletRequest request)
             throws IllegalArgumentException {
-        try {
-            Preconditions.checkNotNull(createdBy, String.format("Header %s needs to be set", JaxrsResource.HDR_CREATED_BY));
-            final Tenant tenant = getTenantFromRequest(request);
-            final UUID tenantId = tenant == null ? null : tenant.getId();
-            final CallContext callContext = contextFactory.createCallContext(accountId, tenantId, createdBy, origin, userType, reason,
-                                                                             comment, getOrCreateUserToken());
-
-            populateMDCContext(callContext);
-
-            return callContext;
-        } catch (final NullPointerException e) {
-            throw new IllegalArgumentException(e.getMessage());
```

Comment:
I think this changes the behavior as the NPE will be returned as 500 while it should be a 400 (user forgot the header):

https://github.com/killbill/killbill/blob/3a52a09da73c4e6d06cdb23272e5c070789ca345/jaxrs/src/main/java/org/killbill/billing/jaxrs/mappers/IllegalArgumentExceptionMapper.java#L39